### PR TITLE
Multi-instance settings

### PR DESCRIPTION
**Pull request by zasca**
#2329

### CODE REVIEW

File: webapp/graphite/settings/__init__.py [hunk 1]
```diff
@@ -207,7 +209,12 @@
 
 ## Load our local_settings
 try:
-  from graphite.local_settings import *  # noqa
+  SETTINGS_MODULE = os.environ['GRAPHITE_SETTINGS_MODULE']
+except:
```
Comment (deniszh) [inline webapp/graphite/settings/__init__.py]: Agreed with Codacy here - please specify exception type here

**Comment by deniszh** [comment]
Agreed with @DanCech - documentation is needed.

**Comment by deniszh** [comment]
@zasca - could you please fix the docs, if you want that change still be included in upcoming 1.1.4 release. Thanks!

**Comment by zasca** [comment]
@deniszh, yeah, I was going to do it, but I ran into regression - setting up a new database and creating the initial schema does not happening with this setting model:
```
# GRAPHITE_SETTINGS_MODULE=graphite.settings.root PYTHONPATH=/opt/graphite/webapp django-admin.py migrate --settings=graphite.settings.root --run-syncdb
STORAGE_DIR: 
2 join(STORAGE_DIR, 'graphite.db'): /opt/graphite/webapp/storage/graphite.db
DATABASES.default: {'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': '/var/lib/graphite/graphite.db', 'HOST': '', 'USER': '', 'PASSWORD': '', 'PORT': ''}}
4 STORAGE_DIR: /opt/graphite/webapp/storage
3 STORAGE_DIR: /opt/graphite/webapp/storage
4 STORAGE_DIR: /opt/graphite/webapp/storage
LAST STORAGE_DIR: /opt/graphite/webapp/storage
Operations to perform:
  Apply all migrations: (none)
Running migrations:
  No migrations to apply.
#
# find / -name graphite.db | wc -l
0
```

**Comment by deniszh** [comment]
I saw that before form time to time but I'm still not getting why it happens.
For newer Django try something like
```
PYTHONPATH=/opt/graphite/webapp django-admin.py makemigrations --settings=graphite.settings
PYTHONPATH=/opt/graphite/webapp django-admin.py migrate auth --settings=graphite.settings
PYTHONPATH=/opt/graphite/webapp django-admin.py migrate --run-syncdb --settings=graphite.settings
```

**Comment by deniszh** [comment]
I'm wondering if moving `settings.py` to `settings/__init__.py` affect all django admin commands. Then we need to change all references in documentation and also old Django commands will stop working after applying this change. I would not apply that to minor release like 1.1.4 then.

**Comment by zasca** [comment]
> I ran into regression - setting up a new database and creating the initial schema does not happening with this setting mode

My bad. **graphite.settings** module was should pass instead:
```
# GRAPHITE_SETTINGS_MODULE=graphite.settings.root \
# PYTHONPATH=/opt/graphite/webapp django-admin.py \
# migrate --settings=**graphite.settings** --run-syncdb
```

**Comment by zasca** [comment]
>I'm wondering if moving `settings.py` to `settings/__init__.py` affect all django admin commands.

@deniszh I've done some tests with debugging and done code review of django tools.
And all looks and working good with suggested settings code.

Yes, potential "out of scopes" errors may occurs, but only when two conditions are met:
1. The level of settings file passed direct to django-admin is NOT a root
2. At the parent modules there are defined variables required for django-admin

```
$ tree settings/
├── __init__.py -> ../orig_settings.py
├── app_settings.py -> ../app_settings.py
└── root.py -> ../local_settings.py

$ cat settings/root.py
SECRET_KEY = 'UNSAFE_DEFAULT'
DATABASES = {'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': '/var/lib/graphite/graphite.db', 'HOST': '', 'USER': '', 'PASSWORD': '', 'PORT': ''}}

$ django-admin migrate --settings graphite.settings.root 2>&1 | grep -v STORAGE
DATABASES.default: {'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': '/var/lib/graphite/graphite.db', 'HOST': '', 'USER': '', 'PASSWORD': '', 'PORT': ''}}
Operations to perform:
  Apply all migrations: (none)
Running migrations:
  No migrations to apply.

$ #########################################################
$ echo "INSTALLED_APPS = ( 'graphite.account', 'graphite.functions' )" >> settings/root.py

$ django-admin migrate --settings graphite.settings.root 2>&1 | grep -v STORAGE
DATABASES.default: {'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': '/var/lib/graphite/graphite.db', 'HOST': '', 'USER': '', 'PASSWORD': '', 'PORT': ''}}
Traceback (most recent call last):
  File "/usr/local/bin/django-admin.py", line 5, in <module>
    management.execute_from_command_line()
  ...
  File "/usr/local/lib/python2.7/dist-packages/django/conf/__init__.py", line 63, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'LOG_ROTATION_COUNT'
````

**Comment by DanCech** [comment]
Is there any reason to push this forward given that we have #2358 ?

**Comment by zasca** [comment]
In my opinion, the location of all settings modules in one package is optimal.
And after moving `settings.py` to `settings/__init__.py` will be possible to put all other settings modules like `app_settings.py` there.

**Comment by DanCech** [comment]
This doesn't seem to be worth the potential issues with backward compatibility.

**Comment by deniszh** [comment]
Same here - I do not see why this is better than #2358
Is #2358 not good enough?

**Pull request closed without merging**